internal/daemonset/daemonset.go

```diff
@@ -0,0 +1,193 @@
+package daemonset
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/test-network-function/cnf-certification-test/internal/clientsholder"
+	"github.com/test-network-function/cnf-certification-test/pkg/provider"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	imageWithVersion = "quay.io/testnetworkfunction/debug-partner:latest"
+	timeout          = 5 * time.Minute
+	nodeExporter     = "node-exporter"
+	containerName    = "container-00"
+	debug            = "debug"
+)
+
+// Delete daemon set
+func DeleteDaemonSet(daemonSetName, namespace string) error {
+	logrus.Infof("Deleting daemon set %s", daemonSetName)
+	deletePolicy := metav1.DeletePropagationForeground
+	client := clientsholder.GetClientsHolder().K8sClient
+	err := client.AppsV1().DaemonSets(namespace).Delete(context.TODO(), daemonSetName, metav1.DeleteOptions{
+		PropagationPolicy: &deletePolicy})
+	if err != nil {
+		return fmt.Errorf("daemonset %s deletion failed: %w", daemonSetName, err)
+	}
+	for start := time.Now(); time.Since(start) < timeout; {
```

Comment:
You should also add some code to handle the "timeout" scenario. Imagine that one of the debug pods do not stop and len(pods.Items) != 0 on each iteration until the timeout.

Reply:
In this case I will do :

time.Sleep(time.Duration(daemonsetDeletionCheckRetryInteval))

when , daemonsetDeletionCheckRetryInteval = 5 sec

@greyerof